src/main/java/org/opensearch/knn/index/codec/KNNCodecVersion.java

```diff
@@ -110,9 +112,24 @@ public enum KNNCodecVersion {
             .knnVectorsFormat(new KNN990PerFieldKnnVectorsFormat(Optional.ofNullable(mapperService)))
             .build(),
         KNN990Codec::new
+    ),
+
+    V_9_12_0(
+        "KNN990Codec",
```

Comment:
Update?

Reply:
Do we need to update SPI as well? https://github.com/opensearch-project/k-NN/blob/main/src/main/resources/META-INF/services/org.apache.lucene.codecs.Codec

Reply:
Yeah thanks I missed it. Somehow as the codec name was different the tests didn't fail.